metadata_updater/metadata_updater.py

```diff
@@ -0,0 +1,333 @@
+# LICENSE file for more information.
+#
+################################################################################
+
+import koordinates
+import os
+import sys
+import fileinput
+import yaml
+import re
+import requests
+import logging
+
+try:
+    import log
+except:
+    from . import log
+
+logger = logging.getLogger(__name__)
+ERRORS = 0
+
+class ConfigReader():
+    """
+    Create a config object that can
+    be passed to the scripts methods
+    """
+
+    def __init__(self):
+        # READ CONFIG
+        with open(os.getcwd()  + '/config.yaml', 'r') as f:
+            config = yaml.load(f)
+
+        if "Connection" in config:
+            self.api_key = config["Connection"]["Api_key"]
+            self.domain = config["Connection"]["Domain"]
+        else:
+
+            raise SystemExit('CONFIG ERROR: No "Connection" section')
+
+        if "Text" in config:
+            self.text_update = config["Text"]["Update"]
+        else:
+            raise SystemExit('CONFIG ERROR: No "Text" section')
+
+        if "Output" in config:
+            self.destination_dir = config["Output"]["Destination"]
+        else:
+            self.destination_dir = os.getcwd()  + '/'
+
+        if "Datasets" in config:
+            self.layers = config["Datasets"]["Layers"]
+            self.filter = config["Datasets"]["Filter"]
+        else:
+            raise SystemExit('CONFIG ERROR: No "Datasets" section')
+
+        if "Test" in config:
+            self.test_dry_run = config["Test"]["Dry_run"]
+            if self.test_dry_run not in (True, False):
+                raise SystemExit('CONFIG ERROR: "Test Dry Run" must be ' \
+                '"True" or "False". Got:"{}" instead'.format(self.test_dry_run))
+
+            self.test_overwrite = config["Test"]["Overwrite_files"]
+            if self.test_overwrite not in (True, False):
+                raise SystemExit('CONFIG ERROR: "test_overwrite" must be ' \
+                '"True" or "False". Got:"{}" instead'.format(self.test_overwrite))
+        else:
+            raise SystemExit('CONFIG ERROR: No "Test" section')
+
+def post_metadata(draft, file):
+    """
+    Update the Data Service draft version for the 
+    layer with edited metadata
+    """
+
+    global ERRORS
+
+    try:
+        xml = open(file).read()
+        draft.set_metadata(xml.encode('utf-8'))
+        return True
+    except koordinates.exceptions.ServerError as e:
+        if 'Expected 204 response, got 200' in str(e):
+            #This is a known bug. - Bug#3
+            return True
+        else:
+            ERRORS += 1
+            logger.critical('metadata update for {0} fail with {1}'.format(draft.id,
+                                                                            str(e)))
+            return False
+
+def get_metadata(layer, config):
+    """
+    Download the layers metadata file
+    """
+
+    file_destination = os.path.join(config.destination_dir,'{0}_{1}.iso.xml'.format(layer.type, 
```

Comment:
Can we also add 'layer.name' to the xml file name? Data Managers will need to read the file name and be able to understand it.

Reply:
Done. 

files now in the format 
`layer_93661_POST_TEST-Weed|Kelp-polygons-(Hydro,-1:4k---1:22k).iso.xml`

Reply:
That looks quite ugly eh? Oh well, still good to have the full name.